src/manifest/update_periods.ts

```diff
@@ -101,3 +65,66 @@ export default function updatePeriods(
     oldPeriods.push(...remainingNewPeriods);
   }
 }
+
+/**
+ * Update old periods by adding new periods and removing
+ * not available ones.
+ * @param {Array.<Object>} oldPeriods
+ * @param {Array.<Object>} newPeriods
+ */
+export function updatePeriods(
```

Comment:
It should work on common cases. I see one case where i am bothered :

Old array:
[ 1 | c | x | 2 | d | 3 ]
New array:
[ 1 | x | 2 | 3 ]
Result:
[ 1 | c | x | 2 | d | 3 ]

It seems that in that case, the c and d periods will not be removed. But,
if i remove the 2 from the old array :

Old array:
[ 1 | c | x | d | 3 ]
New array:
[ 1 | x | 2 | 3 ]
Result:
[ 1 | c | x | 2 | 3 ]

The d is removed. Shouldn't we always remove old periods in between ?

Reply:
OK. I let you do the trick